test/main_test.go

```diff
@@ -0,0 +1,34 @@
+//go:build e2e
+
+package test
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/test/activegate"
+	"github.com/Dynatrace/dynatrace-operator/test/cloudnative"
+	"github.com/Dynatrace/dynatrace-operator/test/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/environment"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/namespace"
+	"github.com/Dynatrace/dynatrace-operator/test/oneagent"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+)
+
+var testEnvironment env.Environment
+
+func TestMain(m *testing.M) {
+	testEnvironment = environment.Get()
+	testEnvironment.BeforeEachTest(dynakube.DeleteDynakubeIfExists())
+	testEnvironment.BeforeEachTest(oneagent.WaitForDaemonSetPodsDeletion())
+	testEnvironment.BeforeEachTest(namespace.Recreate(dynakube.DynatraceNamespace))
+	cloudnative.BeforeEachTest(&testEnvironment)
+	activegate.BeforeEachTest(&testEnvironment)
+
+	testEnvironment.AfterEachTest(dynakube.DeleteDynakubeIfExists())
+	testEnvironment.AfterEachTest(oneagent.WaitForDaemonSetPodsDeletion())
+	testEnvironment.AfterEachTest(namespace.Delete(dynakube.DynatraceNamespace))
+	cloudnative.AfterEachTest(&testEnvironment)
+	activegate.AfterEachTest(&testEnvironment)
```

Comment:
I think it would make more sense for the packages to return a function that fits into `AfterEachTest`.
E.g.:
```suggestion
    testEnvironment.AfterEachTest(cloudnative.Teardown)
    testEnvironment.AfterEachTest(activegate.Teardown)
```